logging/src/main/java/tech/pegasys/teku/logging/ValidatorLogger.java

```diff
@@ -88,9 +89,36 @@ public void aggregationSkipped(final UnsignedLong slot) {
             Color.YELLOW));
   }
 
-  public void producedInvalidAttestation(final UnsignedLong slot, final String reason) {
-    log.error(
-        print(PREFIX + "Produced invalid attestation for slot " + slot + ": " + reason, Color.RED));
+  public void producedInvalidAttestation(
+      final UnsignedLong slot,
+      final int expectedValidatorIndex,
+      final Optional<Integer> actualValidatorIndex,
+      final String reason) {
+    actualValidatorIndex.ifPresentOrElse(
+        actualIndex ->
+            log.error(
+                print(
+                    PREFIX
+                        + "Produced invalid attestation for slot "
+                        + slot
+                        + " with expected validator index: "
+                        + expectedValidatorIndex
+                        + " and actual validator index: "
+                        + actualIndex
+                        + " invalid reason: "
+                        + reason,
+                    Color.RED)),
```

Comment:
Could make this more compact with something like:

```suggestion
    final String actualDescription = actualValidatorIndex.map(idx -> " and actual validator index: " + actualIndex).orElse("");
    log.error(
                print(
                    PREFIX
                        + "Produced invalid attestation for slot "
                        + slot
                        + " with expected validator index: "
                        + expectedValidatorIndex
                        + actualDescription
                        + " . Invalid reason: "
                        + reason,
                    Color.RED)),
```

Reply:
This makes sense to me.